pkg/clients/dynatrace/dynatrace_client_test.go

```diff
@@ -89,6 +90,39 @@ func TestGetResponseOrServerError(t *testing.T) {
 		require.NoError(t, err)
 		assert.NotNil(t, body, "response body available")
 	})
+
+	t.Run("non-JSON response exceeding character limit", func(t *testing.T) {
+		response := []byte(strings.Repeat("really long response", 300))
+		code := http.StatusForbidden
+
+		err := dc.handleErrorResponseFromAPI(response, code, http.Header{})
+		require.Error(t, err)
+
+		maxLen := 1000
+
+		var shortenedResponse []byte
+
+		if len(response) >= maxLen {
+			shortenedResponse = response[:maxLen]
+		} else {
+			shortenedResponse = response
+		}
+
+		assert.EqualError(t, err, fmt.Sprintf("Server returned status code %d; can't unmarshal response (content-type: unknown): %s", code, shortenedResponse))
+	})
+
+	t.Run("HTML response with proxy header set", func(t *testing.T) {
```

Comment:
i would add json response but without header set test case.

Reply:
Done in [d1043d1](https://github.com/Dynatrace/dynatrace-operator/pull/4955/commits/d1043d139156f5af06769057758719153f7ac7f0)